File: PHP/class_base.php
<?php

/*
 * class based fizzbuzz for PHP 5
 */

  class FizzBuzz
  {
      function __construct($x) {
        $this->number = $x;
      }

      public function fizzbuzz() {
        if ($this->number % 15 == 0) {
            $this->_fizzbuzz();
        } else if ($this->number % 5 == 0) {
            $this->_buzz();
        } else if ($this->number % 3 == 0) {
            $this->_fizz();
        } else {
            echo $this->number;
        }
        echo "\n";
      }
      private function _fizzbuzz() {
          $this->_fizz();
          $this->_buzz();
      }

      private function _fizz() {
          echo "Fizz";
      }

      private function _buzz() {
          echo "Buzz";
      }
  }

for ($i = 0;$i < 101; $i ++) {
    $fizzbuzz = new Fizzbuzz($i);
    $fizzbuzz->fizzbuzz();
}

File: Ruby/class_base.rb
#! /usr/bin/env ruby
# -*- coding: utf-8 -*-
# vim: set et sts=2 sw=2 ts=2 :
#

class FizzBuzz
  def initialize(n)
    @n = n
  end

  def fizzbuzz
    return _fizzbuzz if @n % 15 == 0
    return _buzz if @n % 5 == 0
    return _fizz if @n % 3 == 0
    return @n.to_s
  end

  private
  
  def _fizzbuzz
    _fizz + _buzz
  end

  def _fizz
    "Fizz"
  end
  
  def _buzz
    "Buzz"
  end

end

def main
  for i in 1..100 do
    fb = FizzBuzz.new i
    puts fb.fizzbuzz
  end
end

if __FILE__ == $0 then
  main
end


